common/bitcointree/musig2_test.go

```diff
@@ -21,152 +22,205 @@ const (
 	exitDelay   = 512
 )
 
-var vtxoTreeExpiry = common.RelativeLocktime{Type: common.LocktimeTypeBlock, Value: 144}
+var (
+	vtxoTreeExpiry   = common.RelativeLocktime{Type: common.LocktimeTypeBlock, Value: 144}
+	testTxid, _      = chainhash.NewHashFromStr("49f8664acc899be91902f8ade781b7eeb9cbe22bdd9efbc36e56195de21bcd12")
+	serverPrivKey, _ = secp256k1.GeneratePrivateKey()
+	sweepScript, _   = (&tree.CSVMultisigClosure{
+		MultisigClosure: tree.MultisigClosure{PubKeys: []*secp256k1.PublicKey{serverPrivKey.PubKey()}},
+		Locktime:        vtxoTreeExpiry,
+	}).Script()
+	sweepRoot      = txscript.NewBaseTapLeaf(sweepScript).TapHash()
+	receiverCounts = []int{1, 2, 20, 128}
+)
 
-var testTxid, _ = chainhash.NewHashFromStr("49f8664acc899be91902f8ade781b7eeb9cbe22bdd9efbc36e56195de21bcd12")
+func TestBuildAndSignVtxoTree(t *testing.T) {
+	t.Parallel()
 
-func TestRoundTripSignTree(t *testing.T) {
-	fixtures := parseFixtures(t)
-	for _, f := range fixtures.Valid {
-		// Generate 20 cosigners
-		cosigners := make([]*secp256k1.PrivateKey, 20)
-		cosignerPubkeys := make([]*btcec.PublicKey, 20)
-		for i := 0; i < 20; i++ {
-			prvkey, err := secp256k1.GeneratePrivateKey()
+	for _, tc := range generateTestCases(t) {
+		t.Run(tc.name, func(t *testing.T) {
+			_, sharedOutputAmount, err := bitcointree.CraftSharedOutput(
```

Comment:
let's not ignore any return value and also ensure they are valid (non nil/zero/empty)

Reply:
please apply this change to the rest of the test too.